src/util.ts

```diff
@@ -15,3 +17,51 @@ export function arrayEquals<T>(arr1: Array<T>, arr2: Array<T>): boolean {
 export function isPermittedOrigin(url: URL): boolean {
   return url.protocol == 'https:' && arrayEquals(url.hostname.split('.').slice(-2), ['mongodb', 'com']);
 }
+
+function convertTitleCase(name: string): string {
+  return name.replace(/^[_-]*(.)|[_-]+(.)/g, (s, c, d) => (c ? c.toUpperCase() : ' ' + d.toUpperCase()));
+}
+
+/**
+ *
+ * @param taxonomy
+ * @returns a trie structure of taxonomy.
+ * each node is denoted by a 'name' attribute.
+ * other attributes denotes a new node
+ * ['name' and 'display_name' attributes denote name(s) of facet]
+ * [versions have special boolean attribute of 'stable']
+ */
+export function convertTaxonomyResponse(taxonomy: Taxonomy): FacetDisplayNames {
+  const res: FacetDisplayNames = {};
+
+  function addToRes(entityList: TaxonomyEntity[], ref: { [key: string]: any }, property: string) {
+    const conversion = property === 'versions' ? (s: string) => s : convertTitleCase;
```

Comment:
I would strongly consider moving this logic into `convertTitleCase` so we don't have a dynamic function definition - can be hard to trace and isn't very obvious on first glance

Reply:
good catch! looking back, reads like an english sentence 😂

Reply:
done